src/App.tsx

```diff
@@ -1,14 +1,75 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo, TodoStatusFilter } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isTodosLoading, setIsTodosLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [statusFilter, setStatusFilter] = useState<TodoStatusFilter>(
+    TodoStatusFilter.All,
+  );
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setIsTodosLoading(true);
+    getTodos()
+      .then(setTodos)
+      .finally(() => setIsTodosLoading(false));
+  }, []);
+
+  const handleSelectTodo = (todo: Todo) => {
+    setSelectedTodo(todo);
+  };
+
+  const handleCloseTodo = () => {
+    setSelectedTodo(null);
+  };
+
+  const handleFilterChange = (status: TodoStatusFilter) => {
+    setStatusFilter(status);
+  };
+
+  const handleQueryChange = (newQuery: string) => {
```

Comment:
```suggestion
  const handleQueryChange = (event) => {
```
This function now makes no sense - it's just a redundant wrapper
you can change it and just use it instead of: `event => onQueryChange(event.target.value)` in TodoFilter

Your function should accept the event as an argument